src/components/i18nTextarea.vue

```diff
@@ -212,9 +200,24 @@ export default {
 			}
 			this.addTab(lang)
 		},
+		"$store.state.languages": function(languages) {
+			for (const lang in languages) {
+				if (languages[lang]) {
+					this.addTab(lang)
+				}
+			}
+		},
+
 	},
 	created() {
 		this.state = this.value || {}
+		const languages = this.$store.state.languages
```

Comment:
and here

Reply:
The one we have mentioned in created will get executed at the time of component gets created. However there are other components in other pages which are expecting to have their section also gets populated whenever the language is updated in store.

Reply:
it is a very bad pattern to copy-paste code around the places. this must be changed.

Reply:
Yea, Agree. I can create a method for populating languages. which can be called while this component is created and also to watch the language store changes ,whenever there would be any changes in the state

Reply:
extracted functionality to a method